src/averageAges.js

```diff
@@ -55,7 +64,22 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const women = people.filter(person => person.sex === 'f');
+
+  const children = people.filter(person =>
+    women.some(mother => mother.name === person.mother));
+
+  const filteredChildren = onlyWithSon
+    ? children.filter(child => child.sex === 'm')
+    : children;
+
+  const ageDifference = filteredChildren.map(person =>
+    person.born - people.find(mother => mother.name === person.mother).born);
+
+  const averageAge = ageDifference.reduce((sum, age) =>
```

Comment:
create a separate helper function to calculate average age
you do the same three times